libcloud/utils/misc.py

```diff
@@ -282,3 +292,28 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.__repr__())
+
+
+def retry(timeout, retry_delay, backoff, func, *args, **kwargs):
+    """
+    Retry decorator helps to handle common exception.
+    """
+    timeout = DEFAULT_TIMEOUT if timeout is None else timeout
```

Comment:
I think it would also make sense and make it nicer if we make this function a decorator so it could be used like that:

```python
request_func = retry(timeout, retry_delay, ...)(original function args and kwargs)
retry()
```